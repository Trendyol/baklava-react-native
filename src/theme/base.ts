export const space = [0, 2, 4, 8, 12, 16, 24, 32, 48, 64];

export const fonts = ['System', 'System', 'System', 'System', 'System'];

export const fontSizes = [12, 14, 16, 20, 24, 28, 32];

export const fontWeights = ['300', '400', '500', '600', '700'];

export const lineHeights = [12, 14, 16, 20, 24, 28, 32, 36];

export const radii = {
  none: 0,
  small: 4,
  normal: 6,
  large: 8,
  xlarge: 16,
};

export const colors = {
  white: '#ffffff',
  black: '#000000',
  transparent: 'transparent',
  // Color
  primaryColor: '#F27A1A',
  secondaryColor: '#273142',
  tertiaryColor: '#F1F2F7',
  successColor: '#0BC15C',
  dangerColor: '#FF5043',
  warningColor: '#FFB600',
  alternativeColor: '#5794FF',
  featuredColor: '#8C4EFF',
  // Hover
  primaryHover: '#EF6114',
  secondaryHover: '#0F131A',
  tertiaryHover: '#D5D9E1',
  successHover: '#09A44E',
  dangerHover: '#FF3028',
  warningHover: '#FF9800',
  alternativeHover: '#457EFF',
  featuredHover: '#753EFF',
  // Background
  primaryBackground: '#F1F2F7',
  accentBackground: '#FEF2E8',
  secondaryBackground: '#FFFFFF',
  tertiaryBackground: '#F7F7FA',
  successBackground: '#E7F9EF',
  alertBackground: '#FFEEEC',
  warningBackground: '#FFF8E6',
  infoBackground: '#EEF4FF',
  featuredCBackground: '#F4EDFF',
  // Border Color
  borderColor: '#D5D9E1',
  // Content
  contentPrimary: '#273142',
  contentSecondary: '#6E7787',
  contentTertiary: '#95A1B5',
  contentPassive: '#AFBBCA',
};

export default {
  space,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  radii,
  colors,
};
