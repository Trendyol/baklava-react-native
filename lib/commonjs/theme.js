"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.space = exports.radii = exports.fontSizes = exports.default = exports.colors = void 0;
const space = [0, 2, 4, 8, 12, 16, 24, 32, 48, 64];
exports.space = space;
const fontSizes = [12, 14, 16, 20, 24, 28, 32];
exports.fontSizes = fontSizes;
const radii = {
  none: 0,
  small: 4,
  normal: 6,
  large: 8,
  xlarge: 16
};
exports.radii = radii;
const colors = {
  white: '#ffffff',
  black: '#000000',
  transparent: 'transparent',
  // Color
  primaryColor: '#F27A1A',
  secondaryColor: '#273142',
  tertiaryColor: '#F1F2F7',
  successColor: '#0BC15C',
  dangerColor: '#FF5043',
  warningColor: '#FFB600',
  alternativeColor: '#5794FF',
  featuredColor: '#8C4EFF',
  // Hover
  primaryHover: '#EF6114',
  secondaryHover: '#0F131A',
  tertiaryHover: '#D5D9E1',
  successHover: '#09A44E',
  dangerHover: '#FF3028',
  warningHover: '#FF9800',
  alternativeHover: '#457EFF',
  featuredHover: '#753EFF',
  // Background
  primaryBackground: '#F1F2F7',
  accentBackground: '#FEF2E8',
  secondaryBackground: '#FFFFFF',
  tertiaryBackground: '#F7F7FA',
  successBackground: '#E7F9EF',
  alertBackground: '#FFEEEC',
  warningBackground: '#FFF8E6',
  infoBackground: '#EEF4FF',
  featuredCBackground: '#F4EDFF',
  // Border Color
  borderColor: '#D5D9E1',
  // Content
  contentPrimary: '#273142',
  contentSecondary: '#6E7787',
  contentTertiary: '#95A1B5',
  contentPassive: '#AFBBCA',
  // Gradient
  gradientPurple: ['#8C4EFF', '#5794FF'],
  gradientOrange: ['#FFA964', '#F27A1A']
};
exports.colors = colors;
var _default = {
  space,
  fontSizes,
  radii,
  colors
};
exports.default = _default;
//# sourceMappingURL=theme.js.map