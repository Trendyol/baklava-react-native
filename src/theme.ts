import { createTheme } from '@shopify/restyle';

const theme = createTheme({
  colors: {
    white: '#ffffff',
    black: '#000000',
    transparent: 'transparent',
    // Color
    primaryColor: '#F27A1A',
    secondaryColor: '#273142',
    tertiaryColor: '#F1F2F7',
    successColor: '#0BC15C',
    dangerColor: '#FF5043',
    warningColor: '#FFB600',
    alternativeColor: '#5794FF',
    featuredColor: '#8C4EFF',
    // Hover
    primaryHover: '#EF6114',
    secondaryHover: '#0F131A',
    tertiaryHover: '#D5D9E1',
    successHover: '#09A44E',
    dangerHover: '#FF3028',
    warningHover: '#FF9800',
    alternativeHover: '#457EFF',
    featuredHover: '#753EFF',
    // Background
    primaryBackground: '#F1F2F7',
    accentBackground: '#FEF2E8',
    secondaryBackground: '#FFFFFF',
    tertiaryBackground: '#F7F7FA',
    successBackground: '#E7F9EF',
    alertBackground: '#FFEEEC',
    warningBackground: '#FFF8E6',
    infoBackground: '#EEF4FF',
    featuredCBackground: '#F4EDFF',
    // Border Color
    borderColor: '#D5D9E1',
    // Content
    contentPrimary: '#273142',
    contentSecondary: '#6E7787',
    contentTertiary: '#95A1B5',
    contentPassive: '#AFBBCA',
  },
  spacing: {
    none: 0,
    '4xs': 2,
    '3xs': 4,
    '2xs': 8,
    xs: 12,
    s: 14,
    m: 16,
    l: 20,
    xl: 24,
    '2xl': 32,
    '3xl': 40,
    '4xl': 48,
    '5xl': 64,
    '6xl': 80,
  },
  borderRadii: {
    none: 0,
    xs: 2,
    s: 4,
    m: 6,
    l: 8,
    xl: 16,
  },
  breakpoints: {
    phone: 0,
    tablet: 768,
  },
  fonts: {
    light: 'System',
    regular: 'System',
    medium: 'System',
    semiBold: 'System',
    bold: 'System',
  },
  textVariants: {
    defaults: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
      color: 'contentPrimary',
    },
    // Heading
    heading1: {
      fontSize: 32,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 36,
    },
    heading2: {
      fontSize: 28,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 32,
    },
    heading3: {
      fontSize: 24,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 28,
    },
    // Subtitle01
    subtitle01Regular: {
      fontSize: 20,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 24,
    },
    subtitle01Medium: {
      fontSize: 20,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 24,
    },
    subtitle01Semibold: {
      fontSize: 20,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 24,
    },
    subtitle01Bold: {
      fontSize: 20,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 24,
    },
    // Subtitle02
    subtitle02Regular: {
      fontSize: 16,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 20,
    },
    subtitle02Medium: {
      fontSize: 16,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 20,
    },
    subtitle02Semibold: {
      fontSize: 16,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 20,
    },
    subtitle02Bold: {
      fontSize: 16,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 20,
    },
    // Subtitle03
    subtitle03Regular: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    subtitle03Medium: {
      fontSize: 14,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 16,
    },
    subtitle03Semibold: {
      fontSize: 14,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 16,
    },
    subtitle03Bold: {
      fontSize: 14,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 16,
    },
    // Subtitle04
    subtitle04Regular: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 14,
    },
    subtitle04Medium: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 14,
    },
    subtitle04Semibold: {
      fontSize: 12,
      fontWeight: '600',
      fontFamily: 'semiBold',
      lineHeight: 14,
    },
    subtitle04Bold: {
      fontSize: 12,
      fontWeight: '700',
      fontFamily: 'bold',
      lineHeight: 14,
    },
    // Body
    bodyText: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    bodyUnderline: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
      textDecorationLine: 'underline',
    },
    bodyTextLink: {
      fontSize: 14,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 16,
      textDecorationLine: 'underline',
    },
    bodyLongText: {
      fontSize: 14,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 16,
    },
    // Caption
    captionText: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 12,
    },
    captionMedium: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 12,
    },
    captionLongText: {
      fontSize: 12,
      fontWeight: '400',
      fontFamily: 'regular',
      lineHeight: 12,
    },
    captionTextLink: {
      fontSize: 12,
      fontWeight: '500',
      fontFamily: 'medium',
      lineHeight: 12,
      textDecorationLine: 'underline',
    },
  },
});

export type Theme = typeof theme;
export default theme;
