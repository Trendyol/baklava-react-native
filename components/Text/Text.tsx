import React from 'react';
import { Text as T } from 'react-native';
import styled from 'styled-components/native';
import {
  compose,
  color as styledColor,
  space,
  typography,
  position,
  size,
  fontFamily,
  fontWeight,
  textStyle,
  variant as styledVariant,
  fontSize,
} from 'styled-system';
import theme from '../../src/theme';
import { TextVariantTypes } from './types';

const StyledText = styled(T)(
  compose(
    styledColor,
    space,
    typography,
    position,
    size,
    fontSize,
    fontWeight,
    fontFamily,
    textStyle,
    styledVariant({ variants: theme.textStyle }),
  ),
);

function Text({
  variant = 'bodyText',
  color = 'contentPrimary',
  ...rest
}: {
  variant?: TextVariantTypes;
  color?: string;
  [key: string]: any;
}) {
  return (
    <StyledText
      //@ts-ignore
      variant={variant}
      color={color}
      {...rest}>
      {rest.children}
    </StyledText>
  );
}

export default Text;
