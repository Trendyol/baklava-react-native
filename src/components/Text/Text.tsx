import { createText } from '@shopify/restyle';
import { Theme } from '../../theme';

const Text = createText<Theme>();

export default Text;
