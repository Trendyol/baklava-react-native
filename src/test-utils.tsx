import { ThemeProvider } from '@ergenekonyigit/restyle';
import { render } from '@testing-library/react-native';
import * as React from 'react';
import {
  GestureResponderHandlers,
  PanResponder,
  PanResponderCallbacks,
} from 'react-native';
import theme from './theme';
import { PortalProvider } from '.';

theme.fonts = {
  light: 'Rubik-Light',
  regular: 'Rubik-Regular',
  medium: 'Rubik-Medium',
  semiBold: 'Rubik-SemiBold',
  bold: 'Rubik-Bold',
};

export interface RenderOptions {
  wrapper?: React.ComponentType<any>;
  createNodeMock?: (element: React.ReactElement<any>) => any;
}

const AllProviders = ({ children }: { children?: React.ReactNode }) => (
  <ThemeProvider theme={theme}>
    <PortalProvider>{children}</PortalProvider>
  </ThemeProvider>
);

const customRender = (ui: React.ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options }) as any;

const mockPanResponder = () => {
  jest
    .spyOn(PanResponder, 'create')
    .mockImplementation(
      ({
        onStartShouldSetPanResponderCapture,
        onMoveShouldSetPanResponder,
        onPanResponderGrant,
        onPanResponderMove,
        onPanResponderRelease,
      }: PanResponderCallbacks) => ({
        panHandlers: {
          onStartShouldSetResponderCapture: onStartShouldSetPanResponderCapture,
          onMoveShouldSetResponder: onMoveShouldSetPanResponder,
          onResponderGrant: onPanResponderGrant,
          onResponderMove: onPanResponderMove,
          onResponderRelease: onPanResponderRelease,
        } as GestureResponderHandlers,
      }),
    );
};

export * from '@testing-library/react-native';
export { customRender as render, mockPanResponder };
