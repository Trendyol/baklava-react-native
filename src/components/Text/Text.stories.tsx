import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react-native';
import Text from './Text';
import theme from '../../theme';

const variantList = Object.keys(theme.textVariants);

const TextMeta: ComponentMeta<typeof Text> = {
  title: 'Text',
  component: Text,
  argTypes: {
    variant: {
      options: variantList,
      control: { type: 'radio' },
    },
  },
  args: {
    variant: variantList[0] as any,
  },
};

export default TextMeta;

type TextStory = ComponentStory<typeof Text>;

export const Basic: TextStory = args => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text
    </Text>
    <Text variant={args.variant} p="2xs">
      Sample Text
    </Text>
  </>
);

export const Heading: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Heading
    </Text>
    <Text variant="heading1" p="2xs">
      Heading 1
    </Text>
    <Text variant="heading2" p="2xs">
      Heading 2
    </Text>
    <Text variant="heading3" p="2xs">
      Heading 3
    </Text>
  </>
);

export const Subtitle1: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Subtitle 1
    </Text>
    <Text variant="subtitle1Regular" p="2xs">
      Subtitle 1 / Regular
    </Text>
    <Text variant="subtitle1Medium" p="2xs">
      Subtitle 1 / Medium
    </Text>
    <Text variant="subtitle1Semibold" p="2xs">
      Subtitle 1 / Semibold
    </Text>
    <Text variant="subtitle1Bold" p="2xs">
      Subtitle 1 / Bold
    </Text>
  </>
);

export const Subtitle2: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Subtitle 2
    </Text>
    <Text variant="subtitle2Regular" p="2xs">
      Subtitle 2 / Regular
    </Text>
    <Text variant="subtitle2Medium" p="2xs">
      Subtitle 2 / Medium
    </Text>
    <Text variant="subtitle2Semibold" p="2xs">
      Subtitle 2 / Semibold
    </Text>
    <Text variant="subtitle2Bold" p="2xs">
      Subtitle 2 / Bold
    </Text>
  </>
);

export const Subtitle3: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Subtitle 3
    </Text>
    <Text variant="subtitle3Regular" p="2xs">
      Subtitle 3 / Regular
    </Text>
    <Text variant="subtitle3Medium" p="2xs">
      Subtitle 3 / Medium
    </Text>
    <Text variant="subtitle3Semibold" p="2xs">
      Subtitle 3 / Semibold
    </Text>
    <Text variant="subtitle3Bold" p="2xs">
      Subtitle 3 / Bold
    </Text>
  </>
);

export const Subtitle4: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Subtitle 4
    </Text>
    <Text variant="subtitle4Regular" p="2xs">
      Subtitle 4 / Regular
    </Text>
    <Text variant="subtitle4Medium" p="2xs">
      Subtitle 4 / Medium
    </Text>
    <Text variant="subtitle4Semibold" p="2xs">
      Subtitle 4 / Semibold
    </Text>
    <Text variant="subtitle4Bold" p="2xs">
      Subtitle 4 / Bold
    </Text>
  </>
);

export const Body: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Body
    </Text>
    <Text variant="body1" p="2xs">
      Body 1
    </Text>
    <Text variant="body2" p="2xs">
      Body 2
    </Text>
    <Text variant="body3" p="2xs">
      Body 3
    </Text>
    {/* <Text variant="bodyUnderline" p="2xs">
      Body / Underline
    </Text>
    <Text variant="bodyTextLink" p="2xs">
      Body / Text Link
    </Text>
    <Text variant="bodyLongText" p="2xs">
      Body / Long Text
    </Text> */}
  </>
);

export const Caption: TextStory = () => (
  <>
    <Text p="2xs" variant="subtitle1Bold">
      Text Caption
    </Text>
    <Text variant="captionText" p="2xs">
      Caption / Text
    </Text>
  </>
);
