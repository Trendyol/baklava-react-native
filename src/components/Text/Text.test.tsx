import React from 'react';
import theme from '../../theme';
import { render } from '../../test-utils';
import Text from './Text';

describe('Text', () => {
  test('should render text correctly', () => {
    // when
    const { toJSON } = render(
      <Text accessibilityLabel="text">testtesttest</Text>,
    );

    // then
    expect(toJSON()).toMatchSnapshot();
  });

  test('should render given text correctly', () => {
    // when
    const { getByTestId } = render(<Text testID="text">Test Text</Text>);
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.children).toBe('Test Text');
  });

  // Heading
  test('should render heading1 variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="heading1">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(32);
    expect(textComponent.props.style[0].fontWeight).toBe('400');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render heading2 variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="heading2">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(28);
    expect(textComponent.props.style[0].fontWeight).toBe('400');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  test('should render heading3 variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="heading3">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then

    expect(textComponent.props.style[0].fontSize).toBe(24);
    expect(textComponent.props.style[0].fontWeight).toBe('400');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  // Subtitle01
  test('should render subtitle01Regular variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle01Regular">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(20);
    expect(textComponent.props.style[0].fontWeight).toBe('400');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle01Medium variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle01Medium">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(20);
    expect(textComponent.props.style[0].fontWeight).toBe('500');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle01Semibold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle01Semibold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(20);
    expect(textComponent.props.style[0].fontWeight).toBe('600');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle01Bold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle01Bold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(20);
    expect(textComponent.props.style[0].fontWeight).toBe('700');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  // Subtitle02
  test('should render subtitle02Regular variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle02Regular">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(16);
    expect(textComponent.props.style[0].fontWeight).toBe('400');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle02Medium variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle02Medium">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(16);
    expect(textComponent.props.style[0].fontWeight).toBe('500');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle02Semibold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle02Semibold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(16);
    expect(textComponent.props.style[0].fontWeight).toBe('600');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle02Bold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle02Bold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(16);
    expect(textComponent.props.style[0].fontWeight).toBe('700');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  // Subtitle03
  test('should render subtitle03Regular variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle03Regular">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(14);
    expect(textComponent.props.style[0].fontWeight).toBe('400');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle03Medium variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle03Medium">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(14);
    expect(textComponent.props.style[0].fontWeight).toBe('500');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle03Semibold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle03Semibold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(14);
    expect(textComponent.props.style[0].fontWeight).toBe('600');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle03Bold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle03Bold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(14);
    expect(textComponent.props.style[0].fontWeight).toBe('700');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  // Subtitle04
  test('should render subtitle04Regular variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle04Regular">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(12);
    expect(textComponent.props.style[0].fontWeight).toBe('400');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle04Medium variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle04Medium">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(12);
    expect(textComponent.props.style[0].fontWeight).toBe('500');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle04Semibold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle04Semibold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(12);
    expect(textComponent.props.style[0].fontWeight).toBe('600');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render subtitle04Bold variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="subtitle04Bold">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(12);
    expect(textComponent.props.style[0].fontWeight).toBe('700');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  // Body
  test('should render bodyText variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="bodyText">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(14);
    expect(textComponent.props.style[0].fontWeight).toBe('400');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render bodyUnderline variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="bodyUnderline">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(14);
    expect(textComponent.props.style[0].fontWeight).toBe('400');
    expect(textComponent.props.style[0].textDecorationLine).toBe('underline');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render bodyTextLink variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="bodyTextLink">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(14);
    expect(textComponent.props.style[0].fontWeight).toBe('500');
    expect(textComponent.props.style[0].textDecorationLine).toBe('underline');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render bodyLongText variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="bodyLongText">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(14);
    expect(textComponent.props.style[0].fontWeight).toBe('400');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });

  // Caption
  test('should render captionText variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="captionText">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(12);
    expect(textComponent.props.style[0].fontWeight).toBe('400');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render captionMedium variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="captionMedium">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(12);
    expect(textComponent.props.style[0].fontWeight).toBe('500');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render captionLongText variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="captionLongText">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(12);
    expect(textComponent.props.style[0].fontWeight).toBe('400');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  test('should render captionTextLink variant correctly', () => {
    // when
    const { getByTestId } = render(
      <Text testID="text" variant="captionTextLink">
        Test Text
      </Text>,
    );
    const textComponent = getByTestId('text');
    // then
    expect(textComponent.props.style[0].fontSize).toBe(12);
    expect(textComponent.props.style[0].fontWeight).toBe('500');
    expect(textComponent.props.style[0].textDecorationLine).toBe('underline');
    expect(textComponent.props.style[0].color).toBe(
      theme.colors.contentPrimary,
    );
  });
  //
});
