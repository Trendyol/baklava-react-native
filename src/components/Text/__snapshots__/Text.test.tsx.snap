// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render text correctly 1`] = `
<Text
  accessibilityLabel="text"
  accessible={true}
  style={
    Array [
      Object {
        "color": "#273142",
        "fontFamily": "Rubik-Regular",
        "fontSize": 14,
      },
    ]
  }
  testID="text"
>
  testtesttest
</Text>
`;
